homeassistant/components/growatt_server/sensor.py

```diff
@@ -579,91 +711,76 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
 
     devices, plant_id = await hass.async_add_executor_job(get_device_list, api, config)
 
-    entities = []
     probe = GrowattData(api, username, password, plant_id, "total")
-    for sensor in TOTAL_SENSOR_TYPES:
-        entities.append(
-            GrowattInverter(probe, f"{name} Total", sensor, f"{plant_id}-{sensor}")
+    entities = [
+        GrowattInverter(
+            probe,
+            name=f"{name} Total",
+            unique_id=f"{plant_id}-{description.key}",
+            description=description,
         )
+        for description in TOTAL_SENSOR_TYPES
+    ]
 
     # Add sensors for each device in the specified plant.
     for device in devices:
         probe = GrowattData(
             api, username, password, device["deviceSn"], device["deviceType"]
         )
-        sensors = []
+        sensor_descriptions = ()
         if device["deviceType"] == "inverter":
-            sensors = INVERTER_SENSOR_TYPES
+            sensor_descriptions = INVERTER_SENSOR_TYPES
         elif device["deviceType"] == "storage":
             probe.plant_id = plant_id
-            sensors = STORAGE_SENSOR_TYPES
+            sensor_descriptions = STORAGE_SENSOR_TYPES
         elif device["deviceType"] == "mix":
             probe.plant_id = plant_id
-            sensors = MIX_SENSOR_TYPES
+            sensor_descriptions = MIX_SENSOR_TYPES
         else:
             _LOGGER.debug(
                 "Device type %s was found but is not supported right now",
                 device["deviceType"],
             )
 
-        for sensor in sensors:
-            entities.append(
+        entities.extend(
+            [
                 GrowattInverter(
                     probe,
-                    f"{device['deviceAilas']}",
-                    sensor,
-                    f"{device['deviceSn']}-{sensor}",
+                    name=f"{device['deviceAilas']}",
+                    unique_id=f"{device['deviceSn']}-{description.key}",
+                    description=description,
                 )
-            )
+                for description in sensor_descriptions
+            ]
+        )
 
     async_add_entities(entities, True)
 
 
 class GrowattInverter(SensorEntity):
     """Representation of a Growatt Sensor."""
 
-    def __init__(self, probe, name, sensor, unique_id):
+    entity_description: GrowattSensorEntityDescription
+
+    def __init__(
+        self, probe, name, unique_id, description: GrowattSensorEntityDescription
```

Comment:
Please type the whole signature when adding type annotations to a function.

Reply:
@MartinHjelmare can you please provide me with an example of how to do this/where else it's done?
I can submit a PR for the update later (either as part of the PR that was mentioned earlier, or individually)

Reply:
Here's an example:
https://github.com/home-assistant/core/blob/54538bb72bc589e2a415d5e5f0dee4c466e438c0/homeassistant/components/accuweather/sensor.py#L65-L71

Here's the standard library docs for typing:
https://docs.python.org/3/library/typing.html

Reply:
When I rebase our other PR later, I'll update this at the same time as an individual commit.

Reply:
> Please type the whole signature when adding type annotations to a function.

@MartinHjelmare I would prefer not to do that when converting an integration for #53201. There are about 200 more instance left and it would add significantly more effort to each one. Types should ideally be added by someone with knowledge of the integration and dependencies.
Adding the type for `SensorEntityDescription` just helps catch small mistakes easier with the help of Pylance.

Reply:
I don't think we should do partial typing.